### PR TITLE
Fix notice that breaks the plugin if errors are visible

### PR DESCRIPTION
**Pull request by k1sul1**
Removed redundant code and ran phpcbf, which changed some else ifs in a few files.

Also added some comments.

### CODE REVIEW

**Comment by joosev** [comment]
This sets Pakettikauppa shipping method as visible by default. It should not be visible by default.

**Comment by joosev** [comment]
> This sets Pakettikauppa shipping method as visible by default. It should not be visible by default.

This is the reason why it tries to check if there are pakettikauppa shipping methods used that is now removed.

File: includes/class-wc-pakettikauppa-shipping-method.php [hunk 1]
```diff
@@ -555,7 +533,7 @@ private function my_global_form_fields() {
           'show_pakettikauppa_shipping_method' => array(
             'title'   => __('Show Pakettikauppa shipping method', 'wc-pakettikauppa'),
             'type'    => 'select',
-            'default' => 'no',
+            'default' => 'yes',
```
Comment (joosev) [inline includes/class-wc-pakettikauppa-shipping-method.php]: This can't be yes. It has to be no. Pakettikauppa shipping method is only to be used in some special cases and is not meant for everyone to be used.

Comment (k1sul1) [reply includes/class-wc-pakettikauppa-shipping-method.php]: Ok. Why though? As an user, if I install a plugin and complete it's setup wizard, I would expect the plugin to work. With this set to `no`, it doesn't work without first going to Advanced settings under the plugin settings, and setting it to yes:

image

Freshly installed and configured plugin, yet it's not possible to create shipping methods with Pakettikauppa. Am I doing something wrong?

--------

The part that I removed (below) doesn't make much sense to me. If `$this->instance_id` is 0 (which it pretty much always is, correct me if I'm wrong), and `show_pakettikauppa_shipping_method` setting has not been set, it gets all zones and loops through their shipping methods, and if any of them are instances of `WC_Pakettikauppa_Shipping_Method`, it will set the `show_pakettikauppa_shipping_method` setting to `yes`, otherwise `no`, and save the value in the database.

```php
if ( $this->instance_id === 0 ) {
  if ( ! $show_method_set ) {
    $shipping_zones = WC_Shipping_Zones::get_zones();
    $show_pakettikauppa_shipping_method = 'no';

    foreach ( $shipping_zones as $shipping_zone ) {
      foreach ( $shipping_zone['shipping_methods'] as $shipping_object ) {
        if ( get_class($shipping_object) === 'WC_Pakettikauppa_Shipping_Method' ) {
          $show_pakettikauppa_shipping_method = 'yes';
        }
      }
    }

    $this->wc_pakettikauppa_shipment->update_setting('show_pakettikauppa_shipping_method', $show_pakettikauppa_shipping_method);
    $this->wc_pakettikauppa_shipment->save_settings();
    $settings = $this->wc_pakettikauppa_shipment->get_settings();
  }
}
```

a) saving the settings will cause the `Thank you for installing WooCommerce Pakettikauppa! To get started smoothly, please open our setup wizard` notice to potentially disappear before the user has even clicked on it, that surely isn't intended?
b) `get_class($shipping_object) === 'WC_Pakettikauppa_Shipping_Method'` is never true on fresh installs, and the parent if prevents it from running when the install isnt fresh

In other words, the code I removed always sets the `show_pakettikauppa_shipping_method` setting to `no`. If thats's the intended behaviour, I'd like to know why, because it doesn't make sense from the users point of view. Also, as you can see from the screenshot, there's multiple notices originating from that piece of code. These notices have the potential to break admin-ajax.php, rest api and every rss feed that WordPress provides, so they have to be fixed.

Comment (joosev) [reply includes/class-wc-pakettikauppa-shipping-method.php]: instance_id is not always 0. That if is needed, otherwise it ends up in the infinity loop, because get_zones() calls the same code in the end.
--
Pakettikauppa shipping method is not needed for module to work. Pakettikauppa shipping method only is used by merchant who are using it because of historic reasons. Merchants can use any shipping method there is (flat rate for instance). Pakettikauppa is not a traditional shipping method because it is not needed for pricing the shipment, only to extend the functionality of existing shipping methods.

Comment (joosev) [reply includes/class-wc-pakettikauppa-shipping-method.php]: "b) get_class($shipping_object) === 'WC_Pakettikauppa_Shipping_Method' is never true on fresh installs, and the parent if prevents it from running when the install isnt fresh"

Then this needs some kind of a fixing :)

Comment (k1sul1) [reply includes/class-wc-pakettikauppa-shipping-method.php]: Ok, thanks. I'm not too familiar with this yet, I expected it to work like a traditional shipping method. I'll try to find another way around the notice. 

I merged in some new commits from master (that weren't there on monday when I originally cloned the repo), and the setup wizard makes more sense now.

image

When I removed the chunk, this screen didn't work. It listed the zones but mapping wasn't possible, as it didn't list any shipping methods, despite me having multiple shipping methods in the zone.

File: includes/class-wc-pakettikauppa-shipping-method.php [hunk 2]
```diff
@@ -76,14 +76,11 @@ public function __construct( $instance_id = 0 ) {
        */
       public function init() {
         $settings = $this->wc_pakettikauppa_shipment->get_settings();
-        if ( ! isset($settings['show_pakettikauppa_shipping_method']) ) {
-          $show_pakettikauppa_shipping_method = 'yes';
-        } else {
-          $show_pakettikauppa_shipping_method = $settings['show_pakettikauppa_shipping_method'];
-        }
+        $show_method_set = isset($settings['show_pakettikauppa_shipping_method']);
+        $show_pakettikauppa_shipping_method = ! $show_method_set ? 'yes' : $settings['show_pakettikauppa_shipping_method'];
```
Comment (joosev) [inline includes/class-wc-pakettikauppa-shipping-method.php:87]: I think this might have a if (empty($settings)) to set that to 'no'.

Comment (k1sul1) [reply includes/class-wc-pakettikauppa-shipping-method.php:87]: That's would just bloat the code, it doesn't add any value. It's the same code from earlier with less if-else. 

It could be `no` by default, but it doesn't matter. It's going to get refactored anyway.

Comment (joosev) [reply includes/class-wc-pakettikauppa-shipping-method.php:87]: That new code afaik works as following:
is show_pakettikauppa_shipping_method is not set, $show_method_set == false

next one says:
if $show_method_set == false, then set $show_pakettikauppa_shipping_method to 'yes' otherwise use the value in the store.

But if show_pakettikauppa_shipping_method is not set, we have to see if there is any Pakettikauppa shipping methods in use. And if there is, then set it to yes -> otherwise set it to no.

But we can add a check before that to see if $settings is empty, then we can set it as 'no' without checking the shipping methods in the zones.

Comment (joosev) [reply includes/class-wc-pakettikauppa-shipping-method.php:87]: If the settings is empty, then we don't need to save the settings and we can show the wizard.

Comment (k1sul1) [reply includes/class-wc-pakettikauppa-shipping-method.php:87]: Yeah. I'll address that in the refactor.
